app/views/admin_events/show.html.erb

```diff
@@ -82,7 +82,17 @@
   <%= link_to "Delete event", event_path(@event.id), method: :delete,
       data: {confirm: "Are you sure?"}, class: "btn btn-delete" %>
   <% unless @event.approved? %>
-    <%= link_to "Approve event", approve_admin_event_path(@event.id), method: :post,
-      class: "btn btn-save" %>
+  <span class="approve-button">
+  <%= form_for :approve,
+        url: approve_admin_event_path(@event, @event.approved ? 'Unapprove' : 'Approve'),
```

Comment:
same here - ternary operator obsolete -> can just be 
url: approve_admin_event_path(@event,  'Unapprove' ) (...)